spec/01-unit/21-dns-client/02-client_spec.lua

```diff
@@ -559,13 +559,64 @@ describe("[DNS client]", function()
         }, list)
     end)
 
+    it("correctly observes #timeout", function()
+      -- KAG-2300 - https://github.com/Kong/kong/issues/10182
+      -- If we encounter a timeout while talking to the DNS server, expect the total timeout to be close to the
+      -- configured timeout * retrans parameters
+      assert(client.init({
+        resolvConf = {
+          "nameserver 198.51.100.0",
+          "domain one.com",
+        },
+        timeout = 1000,
+        retrans = 1,
```

Comment:
should we test this with retrans > 1 as well and check that the duration is similar to `timeout * retrans`?